### DISCUSSION TITLE
Can't access private GitHub repos in another org

### DISCUSSION THREAD

**Discussion by jmreicha** [renovatebot/renovate#17595] [Request Help]
I am attempting to read and update a private package from another org in my repo but seem to be hitting a wall. Logs say that renovate failed to lookup the dependency.

```
DEBUG: Datasource unknown error
{
  "datasource": "github-tags",
  "packageName": "gruntwork-io/terraform-aws-load-balancer",
  "err": {
    "message": "Cannot read property 'payload' of null",
    "stack": "TypeError: Cannot read property 'payload' of null\n    at CacheableGithubTags.getItems (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/github-releases/cache/cache-base.js:178:107)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async GithubTagsDatasource.getReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/github-tags/index.js:53:29)\n    at async getRegistryReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:57:17)\n    at async fetchReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:240:23)\n    at async getPkgReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:294:34)\n    at async lookupUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/lookup/index.js:52:51)\n    at async fetchDepUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/fetch.js:51:25)\n    at async /home/ubuntu/renovateapp/node_modules/p-map/index.js:57:22"
  }
}
DEBUG: Failed to look up dependency github.com/gruntwork-io/terraform-aws-load-balancer (github.com/gruntwork-io/terraform-aws-load-balancer)(packageFile="module-2/main.tf", dependency="github.com/gruntwork-io/terraform-aws-load-balancer")
```

Here's my current config, following instructions from [here](https://docs.renovatebot.com/getting-started/private-packages/#example-for-gomod):

```
...
"hostRules": [
        {
            "matchHost": "https://api.github.com/repos/gruntwork-io",
            "hostType": "github",
            "encrypted": {
                "token": "ENCRYTPED_TOKEN"
            }
        }
    ],
...
"packageRules": [
       ...
        {
            "managers": [
                "terraform"
            ],
            "additionalBranchPrefix": "{{baseDir}}-",
            "packagePatterns": [
                ".*"
            ]
        }
```

Here's my test case for updating a single module:

```
module "test" {
  source = "git::git@github.com:gruntwork-io/terraform-aws-load-balancer.git//modules/alb?ref=v0.28.0"
}
```

I have tested with curl and the token works to read the private repos.

**Comment by jmreicha** [comment]
I'm wondering if there is something wrong with the credentials, from what I can tell the config is correct. Seeing the following messages in the logs makes me think the creds are okay so not really sure.

```
DEBUG: Found encrypted config
{
  "config": {
    "token": "***********"
  }
}
DEBUG: Trying to decrypt token
DEBUG: Decrypted config using openpgp
DEBUG: Decrypted token
DEBUG: Setting hostRules from config
DEBUG: Adding token authentication for https://api.github.com/repos/gruntwork-io to hostRules
```

It would be nice to be able to see the last few characters of the decrypted secret to help troubleshoot if the correct token is being used.

**Reply by jmreicha** [reply]
Just ran cross https://github.com/renovatebot/renovate/discussions/15978. It sounds like this may be broken currently?

**Reply by rarkins** [reply]
It should hopefully be working thanks to #17179. Can you try removing the `hostType` in your `hostRules`?

**Reply by jmreicha** [reply]
Looks like I'm still getting the same error with `hostType` removed 🤔

**Reply by jmreicha** [reply]
Should I maybe open an issue up for this? I searched around a bit and didn’t see anything with this same error, a few similar ones but those appear to be unrelated.

Would love to get this working either way. This is the only hang up I’ve had, everything else works perfectly.

**Reply by rarkins** [reply]
If you can reproduce it, then yes. Otherwise, this feature is working as far as we know (others have confirmed so).

To reproduce it, you can create new github orgs for this purpose, that way you can add others like myself to the repos to debug it directly.

**Reply by jmreicha** [reply]
Opened an issue https://github.com/renovatebot/renovate/issues/17731.

I noticed a 404 in logs now rather than an error. Hopefully something simple I'm missing.

**Reply by rarkins** [reply]
Unfortunately that's not a reproduction which anyone can fully debug, because the encrypted token is tied to the app. i.e. that token can only be decrypted if it's run in the app environment, and not local debugging.
